orc/orcarm.c

```diff
@@ -1173,7 +1173,7 @@ encode_logical_imm (int size, orc_uint64 val)
   mask = ((orc_uint64)-1ULL) >> (64 - size);
   val &= mask;
 
-  if (mask_ones (val)) {
+  if (mask_shifted_ones (val)) {
```

Comment:
Question:
```
#define mask_ones(val) ((val) && (((val) + 1) & (val)) == 0)
#define mask_shifted_ones(val) ((val) && mask_ones(((val) - 1) | (val)))
```
Could you please describe (as comments in later PRs as well) what do these do? (I don't understand what they are doing)

Reply:
Oh, sorry. I'll append detailed comments related to this for maintenance.

Reply:
Comments are added